crates/transparency/src/map/map.rs

```diff
@@ -43,44 +43,20 @@ use super::proof::Proof;
 ///
 /// ## Hashing Strategy
 ///
-/// Hashes for leaf and branch nodes are calculated using the following input
-/// pattern:
+/// #Leaf nodes
+/// Initially, all leaves have not been inserted into, so their hash value
+/// simply the hash of an empty string.  When a leaf has a value inserted, we
+/// insert the hash of the value prepended with a 0 bit (e.g. 0b0 || Hash(value)
+/// at that node
 ///
-///   1. A single byte prefix determining the type of the node.
-///   2. Zero, one or two values.
-///
-/// ### Leaf Nodes
-///
-/// Leaf node hashes are calculated using the following double-hashing strategy
-/// which ensures protection from concatenation-based collision attacks:
-///
-/// ```hash
-/// H(0xff || H(<key>) || <value>)
-/// ```
-///
-/// ### Branch Nodes
-///
-/// Branch node hashes are calculated using a bit field indicating the presence
-/// of child nodes. For example:
-///
-/// ```hash
-/// // Both children present:
-/// H(0b11 || <left> || <right>)
-///
-/// // Left child present:
-/// H(0b10 || <left>)
-///
-/// // Right child present:
-/// H(0b01 || <right>)
-///
-/// // Empty tree:
-/// H(0b00)
-/// ```
+/// #Branch nodes
+/// The value at a node is the hash of its two children, prepended with a
+/// 1 bit (e.g. 0b1 || Hash(left_val) || Hash(right_val))
```

Comment:
Rephrase branch hash strategy
```suggestion
/// ### Branch Nodes
///
/// Branch node hashes are calculated using the one-byte `1` prefix to prevent collisions with branch hashes.
/// ```hash
/// hash(Branch(left, right)) = hash(0x01 || hash(<left>) || hash(<right>))
/// ```
```